### ISSUE TITLE
Fabric Port?

### ISSUE THREAD

**Issue by Shroototem** [jediminer543/JMT-MCMT#10]
Are there any plans to port this to fabric?

**Comment by jediminer543** [comment]
This is going on the long term enhancement list it'll get the same treatment as #2. I.e. once the current release build is stable and ready I'll work on cross porting it (though I think a back to 1.12.2 port is in order first)

The main issue with this is I've never used fabric before so I'd need to learn how it does it's code transformation and if it can do what I want it to do (as forge gives me a full scriptable api to rewrite entire classes if I need to (modulo issues))

And if anyone wants help replicating MCMT in fabric themselves, the "ask for help and I'll do what I can" of #2 also applies (But I know more about 1.12.2 coremodding than fabric coremodding)

**Comment by KubiRem** [comment]
I think 1.12.2 forge has more priority cuz lots of good mods. Fabric has alot of optimization mods already.

**Comment by Shroototem** [comment]
But we could always use more. XD
I'm trying to get the lowest MSPT possible on my server. Really just for overhead.

**Comment by Titaniumtown** [comment]
This should really get a fabric port!

**Comment by Ytoblo** [comment]
> This should really get a fabric port!

No, just no

**Comment by Titaniumtown** [comment]
@Ytoblo why not?

**Comment by Ytoblo** [comment]
> @Ytoblo why not?

better let a person work on the mod loader on which he is comfortable, with which he is very familiar, and not which "becomes popular". I understand why you need this mod for fabric, you have no experience with forge mappings, and therefore you need fabric for parallel ticks of you server software

**Comment by Ytoblo** [comment]
am i right?

**Comment by Titaniumtown** [comment]
lol

**Comment by KubiRem** [comment]
Yeah U are right @Ytoblo . Forge will be better cuz author knows it. We should first get stable release on modern minecraft versions and then backport to 1.12.2 as somebody sad in another issue. Fabric has alot of optimization mods, and they are enough cuz there are no "big" mods on fabric.

**Comment by himekifee** [comment]
I'd like to help porting it to fabric. I had some fabric experience and willing to know more about Minecraft mechanisms.

**Comment by himekifee** [comment]
I did the first release of my porting. It can be found at [MCMTFabric](https://github.com/himekifee/MCMTFabric). For some reason, it doesn't seem to be working and proper logging and configuration loading is in progress.

**Comment by jediminer543** [comment]
@himekifee 

Oooooh, very Jazzy, I'll try having a look at the code properly tomorrow

One thing I would note is that your code lacks the ability at present to effectively perform the functionality of sync_fu

But props to you for making it work; if you want to talk, please yell; I'm working on setting up a discord server, but DMs can work before then if needs be
